ast/term.go

```diff
@@ -1427,6 +1428,17 @@ func (s *set) Union(other Set) Set {
 	return r
 }
 
+func (s *set) Multiply(other Set) Set {
+	r := NewSet()
+	s.Foreach(func(term1 *Term) {
+		other.Foreach(func(term2 *Term) {
+			el := ArrayTerm(term1, term2)
+			r.Add(el)
+		})
+	})
+	return r
+}
```

Comment:
Since we're adding this to the ast package, it would be nice to have some unit tests added, too.

Alternatively, we could move this whole implementation into the builtin definition, and test that via external yaml cases.

What do you think? I think both approaches would be fine -- but I would appreciate a test case or two in the yaml test case definitions, because they're also used to test the Wasm evaluation features of OPA.